syft/pkg/cataloger/javascript/parse_yarn_lock.go

```diff
@@ -80,15 +80,17 @@ func parseYarnLock(path string, reader io.Reader) ([]*pkg.Package, []artifact.Re
 
 	// check if we have valid unsaved data after end-of-file has reached
 	if currentPackage != noPackage && currentVersion != noVersion && !parsedPackages.Contains(currentPackage+"@"+currentVersion) {
-		packages = append(packages, newYarnLockPackage(currentPackage, currentVersion))
+		pkgs = append(pkgs, newYarnLockPackage(resolver, reader.Location, currentPackage, currentVersion))
 		parsedPackages.Add(currentPackage + "@" + currentVersion)
 	}
 
 	if err := scanner.Err(); err != nil {
 		return nil, nil, fmt.Errorf("failed to parse yarn.lock file: %w", err)
 	}
 
-	return packages, nil, nil
+	pkg.Sort(pkgs)
```

Comment:
Is the main reason for sorting each of these functions for stable results in tests?

Reply:
exactly